Student/Student/Routes.swift

```diff
@@ -16,113 +16,477 @@
+    "/files/folder/:folderID": { _, params in
+        return fileListController(params: params)
+    },
+    "/:context/:contextID/files": { url, params in
+        if let preview = previewFileViewController(url: url, params: params) { return preview }
+        return fileListController(params: params)
+    },
+
+    "/files/folder/*subFolder": { url, params in
+        if let preview = previewFileViewController(url: url, params: params) { return preview }
+        var props = params
+        props["context"] = "users"
+        props["contextID"] = "self"
+        return HelmViewController(moduleName: "/:context/:contextID/files/folder/*subFolder", props: makeProps(url, params: props))
+    },
+    "/:context/:contextID/files/folder/*subFolder": { url, params in
+        if let preview = previewFileViewController(url: url, params: params) { return preview }
+        return HelmViewController(moduleName: "/:context/:contextID/files/folder/*subFolder", props: makeProps(url, params: params))
+    },
+
+    "/:context/:contextID/files/folders/:folderID": { _, params in
+        // We don't support routing to a specific folderID yet, that's to come later
+        return fileListController(params: params)
     },
 
-    RouteHandler(.pages(forCourse: ":courseID"), name: "course_pages") { _, params in
+    "/files/:fileID": fileViewController,
+    "/files/:fileID/download": fileViewController,
+    "/files/:fileID/old": fileViewController,
+    "/:context/:contextID/files/:fileID": { url, params in
+        if params["context"] == "courses", let courseID = params["contextID"],
+            let controller = moduleItemController(for: url, courseID: courseID) {
+            return controller
+        }
+        return fileViewController(url: url, params: params)
+    },
+    "/:context/:contextID/files/:fileID/download": fileViewController,
+
+    "/courses/:courseID/grades": nil,
+
+    "/courses/:courseID/modules": { _, params in
+        guard let courseID = params["courseID"] else { return nil }
+        guard let session = CKIClient.current?.authSession else { return nil }
+        let contextID = ContextID.course(withID: courseID)
+        // Restrict access to Modules tab if it's hidden (unless it is the home tab)
+        let modulesTab = try? Tab.modulesTab(for: contextID, in: session)
+        let homeTab = try? Tab.homeTab(for: contextID, in: session)
+        let modulesAreHome = homeTab != nil && homeTab!.routingURL(session).flatMap { $0.path.contains("/modules") } ?? false
+        if !modulesAreHome, modulesTab?.hidden ?? false {
+            let message = NSLocalizedString("That page has been disabled for this course", comment: "")
+            let alert = UIAlertController(title: nil, message: message, preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: NSLocalizedString("Dismiss", comment: ""), style: .default))
+            return alert
```

Comment:
This is the case I was thinking of where we route to a `UIAlertController`. Turns out this logic is broken on prod and in this PR so I wasn't able to see exactly what happens..

Web redirects to "/courses/:courseID" which is what I think we could do but we can do it later since it's not a regression. But I am a little concerned we route to UIAlertControllers in other places.